storage/src/main/java/tech/pegasys/teku/storage/server/DepositStorage.java

```diff
@@ -52,27 +60,41 @@ public void stop() {}
   }
 
   private ReplayDepositsResult replayDeposits() {
-    isSyncingFromDatabase = true;
+    if (!eth1DepositsFromStorageEnabled) {
+      startingBlock = Optional.of(BigInteger.valueOf(-1L));
+      return new ReplayDepositsResult(UnsignedLong.ZERO, false);
+    }
+
     final DepositSequencer depositSequencer =
         new DepositSequencer(eth1EventsChannel, database.getMinGenesisTimeBlock());
     try (Stream<DepositsFromBlockEvent> eventStream = database.streamDepositsFromBlocks()) {
       eventStream.forEach(depositSequencer::depositEvent);
     }
     depositSequencer.depositsComplete();
-    isSyncingFromDatabase = false;
-    return depositSequencer.done();
+    ReplayDepositsResult result = depositSequencer.done();
```

Comment:
nit: Couldn't we just have `depositsComplete()` return the `ReplayDepositsResult`?